core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java

```diff
@@ -300,17 +302,22 @@ public MainClassBuildItem mainClassBuildStep(BuildProducer<GeneratedClassBuildIt
             mainClassName = quarkusMainAnnotations.get("");
         }
         if (mainClassName.equals(MAIN_CLASS)) {
-            //generate a main that just runs the app, the user has not supplied a main class
-            ClassCreator file = new ClassCreator(new GeneratedClassGizmoAdaptor(generatedClass, true), MAIN_CLASS, null,
-                    Object.class.getName());
+            if (quarkusApplicationClass.isPresent()) {
```

Comment:
what is this fixing ? (I assume its unrelated to picocli support ?)

Reply:
We need this because `MainClassBuildItem` looks for `QuarkusMain` annotated classes in `CombinedIndexBuildItem`. Since `PicocliRunner` is part of picocli extension it won't be visible in index.

Reply:
This makes sense to me